rusty-machine/src/learning/gmm.rs

```diff
@@ -132,6 +132,36 @@ impl GaussianMixtureModel {
         }
     }
 
+    /// Constructs a new GMM with the specified prior mixture weights.
+    ///
+    /// The mixture weights must have the same length as the number of components.
+    /// The sum of the mixture weights must be 1.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rusty_machine::learning::gmm::GaussianMixtureModel;
+    /// use rusty_machine::linalg::vector::Vector;
+    ///
+    /// let mix_weights = Vector::new(vec![0.25, 0.25, 0.5]);
+    ///
+    /// let _ = GaussianMixtureModel::with_weights(3, mix_weights);
+    /// ```
+    pub fn with_weights(k: usize, mixture_weights: Vector<f64>) -> GaussianMixtureModel {
+        assert!(mixture_weights.size() == k, "Mixture weights must have length k.");
```

Comment:
([`PathBuf.to_str`](https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.to_str) returning `None` if the path isn't valid Unicode is an example of something in the standard library returning an Option even when the typical case will almost always be `Some`. I agree that in this case we would want a `Result`.)